components/post-date/index.js

```diff
@@ -0,0 +1,82 @@
+import { __ } from '@wordpress/i18n';
+import { DateTimePicker } from '@wordpress/components';
+import { getSettings, dateI18n } from '@wordpress/date';
+import { useEntityProp } from '@wordpress/core-data';
+import PropTypes from 'prop-types';
+import { usePopover } from '../../hooks/use-popover';
+import { usePost } from '../../hooks';
+
+export const PostDatePicker = ({ date, setDate }) => {
+	const settings = getSettings();
+	// To know if the current time format is a 12 hour time, look for "a".
+	// Also make sure this "a" is not escaped by a "/".
+	const is12Hour = /a(?!\\)/i.test(
+		settings.formats.time
+			.toLowerCase() // Test only for the lower case "a".
+			.replace(/\\\\/g, '') // Replace "//" with empty strings.
+			.split('')
+			.reverse()
+			.join(''), // Reverse the string and test for "a" not followed by a slash.
+	);
+
+	return <DateTimePicker currentDate={date} onChange={setDate} is12Hour={is12Hour} />;
+};
+
+PostDatePicker.propTypes = {
+	date: PropTypes.string.isRequired,
+	setDate: PropTypes.func.isRequired,
+};
+
+export const PostDate = (props) => {
+	const { placeholder = __('No date set', 'tenup'), format = undefined, ...rest } = props;
```

Comment:
Same here with the defaults, in the event of this being translated it would show up two times too

Reply:
Fixed in [c853dff](https://github.com/10up/block-components/pull/178/commits/c853dff4a098d3465b3bd8f0aa3a333ffd29ce64)